File: DTOs/Primitives/Wage.swift
//
//  Wage.swift
//  Created by Алексей on 05.08.2024.
//

import Foundation

public struct Wage: Codable, Hashable {
    public var price: Price
    public var period: PaymentPeriod

    public init(
        price: Price,
        period: PaymentPeriod
    ) {
        self.price = price
        self.period = period
    }
}

File: Node/src/dto/primitives/Wage.ts
import { ValidateNested } from "class-validator"
import { PaymentPeriod } from "../enums/PaymentPeriod"
import { Price } from "../primitives/Price"

export class Wage {
    @ValidateNested()
    price: Price

    period: PaymentPeriod

    constructor(
        price: Price,
        period: PaymentPeriod
    ) {
        this.price = price
        this.period = period
    }
}